neurom/core/_neuron.py

```diff
@@ -131,7 +206,7 @@ def transform(self, trans):
 
     def iter_sections(self, order=Tree.ipreorder):
         '''iteration over section nodes'''
-        return order(self.root_node)
+        return iter_sections(self, iterator_type=order)
```

Comment:
I'm tempted to add iter_sections, iter_segments, iter_neurites to Neuron() and Population(), and iter_segments to Section(), thus making the iterators tab-compleable.

Thoughts?